### PR TITLE
Added template sample 17 to rename a file

### PR DESCRIPTION
**Pull request by ewilliams0305**
per closed issue #1238 a sample has been created to demonstrate renaming a file.

### Problem
No available examples renaming files or demonstrating the `fileRename` attribute .

### Solution
Created a new sample No.17 demonstrating the `fileRename` attribute.

### Checks:
- Sample code only, no testing required.

### CODE REVIEW

**Comment by ewilliams0305** [comment]
@dotnet-policy-service agree [company=""]

**Comment by ewilliams0305** [comment]
@dotnet-policy-service agree

File: dotnet-template-samples/content/17-rename-file/17-rename-file.sln
Comment (MiYanni) [inline dotnet-template-samples/content/17-rename-file/17-rename-file.sln:1]: Solution file isn't necessary for the samples folders.

File: dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            "displayName": "File name",
+            "description": "Renames the default FileToRename.cs to the value provided.",
+            "datatype": "text",
+            "defaultValue": "FileToRename",            
+            "replaces": "FileToRename",                // Optionally renames the text inside all files matching 'FileToRename'
+            "fileRename": "FileToRename"               // Finds files and folders including the text 'FileToRename' and replaces it with the value stored in the 'renameFileParameter'
+        }
+      },
+  "sourceName": "MyProject.Con",              
+  "preferNameDirectory":true
```
Comment (MiYanni) [inline dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json:25]: JSON doesn't officially support comments. Please remove them. Also, here's some whitespace cleanup.

```suggestion
  "$schema": "http://json.schemastore.org/template",
  "author": "Contoso",
  "classifications": [ "Console" ],
  "name": "Contoso Sample 17",
  "identity": "MyProject.17.Sample.CSharp",
  "groupIdentity": "MyProject.17.Sample",
  "shortName": "sample17",
  "tags": {
    "language": "C#",
    "type": "project"
  },
  "symbols": {
    "renameFileParameter": {
      "type": "parameter",
      "displayName": "File name",
      "description": "Renames the default FileToRename.cs to the value provided.",
      "datatype": "text",
      "defaultValue": "FileToRename",
      "replaces": "FileToRename",
      "fileRename": "FileToRename"
    }
  },
  "sourceName": "MyProject.Con",
  "preferNameDirectory": true
```

**Review by MiYanni — CHANGES_REQUESTED** [review]

File: dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json (hunk 1 above)
Comment (ewilliams0305) [reply dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json]: No problem, I just removed and formatted the template.json, comments we included based on the 01-basic-template example [comments](https://github.com/dotnet/templating/blob/main/dotnet-template-samples/content/01-basic-template/MyProject.Con/.template.config/template.json)

Comment (MiYanni) [reply dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json]: Oh, that's interesting. 🤔 I wouldn't use that as a good example. 😅

Comment (ewilliams0305) [reply dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json]: I thought it was odd

**Review by MiYanni — APPROVED** [review]

**Comment by MiYanni** [comment]
Thanks for the contribution!

File: dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json (hunk 1 above)
Comment (sayedihashimi) [reply dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json]: That was probably my fault 😁

**Pull request merged**